[WTF] Align Linux OSAllocator more aligned to the other platforms

Constellation opened the pull request:
#### 8681868332b8b4ed96ac0005a7806b7fad0c11fd
<pre>
[WTF] Align Linux OSAllocator more aligned to the other platforms
<a href="https://bugs.webkit.org/show_bug.cgi?id=243049">https://bugs.webkit.org/show_bug.cgi?id=243049</a>

Reviewed by Adrian Perez de Castro.

This patch aligns Linux OSAllocator behavior more aligned to the other platforms,
to pave the way to using libpas JITHeap.

* Source/WTF/wtf/posix/OSAllocatorPOSIX.cpp:
(WTF::OSAllocator::tryReserveUncommitted):
(WTF::OSAllocator::tryReserveUncommittedAligned):
(WTF::OSAllocator::commit):
(WTF::OSAllocator::decommit):

Canonical link: <a href="https://commits.webkit.org/252739@main">https://commits.webkit.org/252739@main</a>
</pre>

aperezdc commented on Source/WTF/wtf/posix/OSAllocatorPOSIX.cpp:
```diff
@@ -230,14 +230,9 @@ void* OSAllocator::reserveAndCommit(size_t bytes, Usage usage, bool writable, bo
-    if (mprotect(address, bytes, protection))
-        CRASH();
-    madvise(address, bytes, MADV_WILLNEED);
+    UNUSED_PARAM(writable);
+    UNUSED_PARAM(executable);
+    while (madvise(address, bytes, MADV_WILLNEED) == -1 && errno == EAGAIN) { }
```
Given that the `writable` and `executable` parameters were only used under `OS(LINUX)`, I think it would make sense to remove the parameters completely, and apply the same change to `OSAllocatorWin.cpp`—unless there is some reason why that would be an issue on Windows, of course.

Constellation replied:
That sounds good! I'll do it in a follow-up patch :)

aperezdc reviewed (approved):
r=me, with a small nit which you may want to address, maybe in a follow-up patch 😺

Constellation committed:
[WTF] Align Linux OSAllocator more aligned to the other platforms
https://bugs.webkit.org/show_bug.cgi?id=243049

Reviewed by Adrian Perez de Castro.

This patch aligns Linux OSAllocator behavior more aligned to the other platforms,
to pave the way to using libpas JITHeap.

* Source/WTF/wtf/posix/OSAllocatorPOSIX.cpp:
(WTF::OSAllocator::tryReserveUncommitted):
(WTF::OSAllocator::tryReserveUncommittedAligned):
(WTF::OSAllocator::commit):
(WTF::OSAllocator::decommit):

Canonical link: https://commits.webkit.org/252739@main
